controller/source/controller.cpp

```diff
@@ -31,12 +31,13 @@ auto handle_get(const std::unique_ptr<kv_client> &client,
   // if the key exists and complies with the gdpr rules
   // then return the value of the get operation
   if (filter.validate(query_args, def_policy)) {
-    logger->log(query_args, res.has_value());
+    if (filter.check_monitoring()){
+      logger->log(query_args, res.has_value());
```

Comment:
I think it would be better to perform this action inside the validate function because even an invalid access should be log (so that any attempt to access a value can be traced).

Reply:
This is okay with `get`. But the operation is not yet performed for the `put` and `delete` methods during the `validate` call. So, it will block us from logging if put/delete succeeded. Do you suggest that we just log each attempt?

Reply:
In any case I think we should "eagerly" log the attempt, since if we write it in the log after it's performed and there is a crash (for example) we might lose information.
So we could either omit the query result, or simply append it for the `put` and `delete` as a single bit at the end of the entry.
I'd recommend the first for now. Wdyt?

Reply:
Sounds extremely heavy imo.
It would make more sense to have a `complete_logging` function in the logger where the outcome of an operation is added to the respective log entry after it's done.
You can use the `validate` function return value to determine when it's needed.